lib/draftjs_exporter/style_state.rb

```diff
@@ -23,16 +23,34 @@ def text?
     end
 
     def element_attributes
-      return {} unless styles.any?
-      { style: styles_css }
+      css_attributes = calculate_styles
+      if css_attributes.any?
+        { style: css_attributes.join }
+      else
+        {}
+      end
     end
 
-    def styles_css
+    private
+
+    def calculate_styles
       styles.map { |style|
-        style_map.fetch(style)
-      }.inject({}, :merge).map { |key, value|
+        fetch_style(style)
+      }.compact.inject({}, :merge).map { |key, value|
         "#{hyphenize(key)}: #{value};"
-      }.join
+      }
+    end
+
+    def fetch_style(style)
+      style_map.fetch(style) { |style|
+        case unknown_styles
+        when :warn, :ignore
+          warn("Missing definition for style: #{style}") if unknown_styles == :warn
+          nil
+        else
+          raise KeyError, "Cannot find style #{style}"
```

Comment:
Quick question, we weren't raising this before. Is the intention to use `:warn` or `:ignore` as the `unknown_styles` inside of OLS?

Reply:
Warn would be the preferred way. It should get logged. Not that we really care. Just so that it does not raise a KeyError.